uv/reporter/state.py

```diff
@@ -88,6 +88,14 @@ def report_params(m: Dict[str, str]) -> None:
   return get_reporter().report_params(m)
 
 
+def report_artifact(local_path: str) -> None:
```

Comment:
This will fail for any reporter other than MLFlowReporter. I get a little nervous about mirroring more of the MLFlow API into the UV api than we already have. Metrics make total sense. Artifacts, and the model persistence methods, feel out of scope.

What if we mirror this only into the `mlflow.reporter` module? You could either:

- keep it ONLY on the MLFlowReporter, or
- add this global, but have it only do anything if `get_reporter` returns an instance of MLFlowReporter.

That is going to be a little tricky too, because in that case you won't trigger anything if you wrap the MLFlowReporter in anything.

@ajslone , Q for you. Would it be too wild if.... we leaned on Python and overrode `__getattr__` to pass ANY method call down the reporter tree? That way if anything in the stack had a method like `report_artifact` it'd get called, but if not that would get ignored?